### PR TITLE
feat(gpu): Implement unsigned_overflowing_scalar_add

### PR DESCRIPTION
**Pull request by bbarbakadze**
### PR content/description
Implement overflowing scalar add
### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Review by agnesLeroy — COMMENTED** [review]
We should also add the overflowing entry points to the hl api

File: tfhe/benches/integer/bench.rs [hunk 1]
```diff
@@ -1723,6 +1728,11 @@ mod cuda {
         display_name: overflowing_add
     );
 
+    define_cuda_server_key_bench_clean_input_fn!(
```
Comment (agnesLeroy) [inline tfhe/benches/integer/bench.rs:1731]: Same here

**Review by agnesLeroy — COMMENTED** [review]
Thanks Beka, looks good to me but @tmontaigu will know better than me for the first commit that modifies the noise level.

**Comment by tmontaigu** [comment]
I don't think the first commit is right with regards to how the GPU works compared to CPU

On the CPU, when the input to a PBS is a trivial ciphertext, we actually skip the PBS and apply the LUT directly via clear operations, so the output is still a trivial ciphertext without noise
https://github.com/zama-ai/tfhe-rs/blob/e825277219cc142bf13959b486192ddc97711445/tfhe/src/shortint/server_key/mod.rs#L750

I don't think you do that on GPU, meaning that you apply PBS on trivial inputs meaning the output has noise and is no longer a trivial

**Comment by agnesLeroy** [comment]
Ok so we'll remove the first commit and in case the gpu feature is activated we won't check if NoiseLevel is zero in case a trivial ciphertext is given as input to unsigned_overflowing_scalar_add

**Comment by agnesLeroy** [comment]
We just pushed the change

**Comment by tmontaigu** [comment]
Hum ok I looked at the wrong think, I check again

**Review by tmontaigu — APPROVED** [review]

**Pull request merged**